subprojects/core/src/main/java/org/gradle/plugin/use/internal/DefaultPluginRequest.java

```diff
@@ -17,26 +17,27 @@
 package org.gradle.plugin.use.internal;
 
 import org.gradle.groovy.scripts.ScriptSource;
-import org.gradle.plugin.internal.PluginId;
+import org.gradle.plugin.use.PluginId;
 
-public class DefaultPluginRequest implements PluginRequest {
+public class DefaultPluginRequest implements InternalPluginRequest {
 
     private final PluginId id;
     private final String version;
     private final boolean apply;
     private final int lineNumber;
     private final String scriptDisplayName;
+    private final ConfiguredOptions configuredOptions = new ConfiguredOptions();
```

Comment:
We should create a new, mutable plugin request and then make it immutable again once the configuration action is done. The plugin request should not have a mutable nested object.